sake/mario_kart_wii.go

```diff
@@ -0,0 +1,194 @@
+		contentType := strings.Replace(contentType, boundary, quotedBoundary, 1)
+		request.Header.Set("Content-Type", contentType)
+	}
+
+	err = request.ParseMultipartForm(rkgdFileMaxSize)
+	if err != nil {
+		logging.Error(moduleName, "Failed to parse the multipart form")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultFileNotFound))
+		return
+	}
+
+	file, fileHeader, err := request.FormFile(rkgdFileName)
+	if err != nil {
+		logging.Error(moduleName, "Failed to find the ghost file")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultFileNotFound))
+		return
+	}
+	defer file.Close()
+
+	if fileHeader.Size < rkgdFileMinSize || fileHeader.Size > rkgdFileMaxSize {
+		logging.Error(moduleName, "The size of the ghost file is invalid")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultFileTooLarge))
+		return
+	}
+
+	ghostFile := make([]byte, fileHeader.Size)
+	_, err = io.ReadFull(file, ghostFile)
+	if err != nil {
+		logging.Error(moduleName, "Failed to read contents of the ghost file")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultFileTooLarge))
+		return
+	}
+
+	if !isRKGDFileValid(ghostFile) {
+		logging.Error(moduleName, "Received an invalid ghost file")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultFileTooLarge))
+		return
+	}
+
+	err = database.UploadMarioKartWiiGhostFile(pool, ctx, regionId, courseId, score, pid, playerInfo, ghostFile)
+	if err != nil {
+		logging.Error(moduleName, "Failed to insert the ghost file into the database")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultServerError))
+		return
+	}
+
+	responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultSuccess))
+}
+
+func isPlayerInfoValid(playerInfoString string) bool {
+	playerInfoByteArray, err := common.DecodeGameSpyBase64(playerInfoString, common.GameSpyBase64EncodingURLSafe)
+	if err != nil {
+		return false
+	}
+
+	if len(playerInfoByteArray) != playerInfoSize {
+		return false
+	}
+
+	var playerInfo playerInfo
+	reader := bytes.NewReader(playerInfoByteArray)
+	err = binary.Read(reader, binary.BigEndian, &playerInfo)
+	if err != nil {
+		return false
+	}
+
+	if common.RFLCalculateCRC(playerInfo.MiiData[:]) != 0x0000 {
+		return false
+	}
+
+	controllerId := common.MarioKartWiiControllerId(playerInfo.ControllerId)
+
+	return controllerId.IsValid()
+}
+
+func getMultipartBoundary(contentType string) string {
+	startIndex := strings.Index(contentType, "boundary=")
+	if startIndex == -1 {
+		return ""
+	}
+	startIndex += len("boundary=")
+
+	return contentType[startIndex:]
+}
+
+func isRKGDFileValid(rkgdFile []byte) bool {
+	const rkgdFileMagic = 0x524B4744 // RKGD
+
+	if binary.BigEndian.Uint32(rkgdFile[:4]) != rkgdFileMagic {
+		return false
+	}
```

Comment:
You could use `bytes.Equal` here.
```go 
rkgdFileMagic := []byte{'R', 'K', 'G', 'D'}

if !bytes.Equal(rkgdFile[:4], rkgdFileMagic) {
    return false
}
```

Reply:
Addressed in commit [ff8144f](https://github.com/WiiLink24/wfc-server/pull/63/commits/ff8144f89f33683efa7ef7a8e5955574db3452af).